db/internal_stats.h

```diff
@@ -45,6 +45,9 @@ struct DBPropertyInfo {
 
   // @param props Map of general properties to populate
   bool (InternalStats::*handle_map)(std::map<std::string, std::string>* props);
+
+  // special case to get the immutable db statistics
+  bool is_immutable_db_statistic;
```

Comment:
It doesn't create much value to parse to this bool and still do the work in DBImpl::GetProperty() info.
Either we still provide a callback here with type like

bool (DBImpl::*handle_string_dbimpl)(std::string* value);

Or you do the parsing and execution all in DBImpl::GetProperty(), before calling GetPropertyInfo().